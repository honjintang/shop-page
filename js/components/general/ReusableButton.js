import React from 'react'
import Button from 'react-bootstrap/Button'

const ReusableButton = ({ id, label, onClick, active }) =>
    <Button id={id} variant="outline-light" onClick={onClick} active={active}>{label}</Button>

export default ReusableButton