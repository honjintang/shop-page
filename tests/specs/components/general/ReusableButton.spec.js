import React from 'react'
import { shallow } from 'enzyme'
import ReusableButton from '~/components/general/ReusableButton'

describe('components/general/ReusableButton', () => {
    let component
    const mockFn = jest.fn()

    it('matches snapshot', () => {
        component = shallow(<ReusableButton 
                id='mockId'
                label='button label'
                onChange={mockFn}
                active={false}
                 />)
            expect(component).toMatchSnapshot()
    })
})