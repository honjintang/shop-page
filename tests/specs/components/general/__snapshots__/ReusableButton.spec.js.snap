// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/general/ReusableButton matches snapshot 1`] = `
<Button
  active={false}
  disabled={false}
  id="mockId"
  type="button"
  variant="outline-light"
>
  button label
</Button>
`;
